api/src/main/java/org/pmiops/workbench/api/WorkspacesController.java

```diff
@@ -764,6 +758,8 @@ private void writeWorkspaceConfigFile(org.pmiops.workbench.firecloud.model.Works
   }
 
   private String convertClusterName(String workspaceId) {
+    // TODO(calbach): This conversion method is flawed as cluster creation
+    // retries by appending digits to the cluster name. Clean this up.
```

Comment:
Thanks for catching this. We should probably brainstorm at some point how we want to try handling this.

Reply:
Yes. I commented to myself on my doc about this:

https://docs.google.com/a/google.com/document/d/1GbaL1otgswBEOqfte8JO8uQDG_vSEG4a-qcKMEdufLI/edit?disco=AAAABq-ZYFU

Needs some more thought, but with my proposed cluster changes it is conceivable we just use a single stable cluster identifier (perhaps as a function of environment as well); attaching to that cluster if we see it and reviving it if it enters a bad state.